File: 2240 Number of Ways to Buy Pens and Pencils/2240number-of-ways-to-buy-pens-and-pencils.py
class Solution:
    def waysToBuyPensPencils(self, total: int, cost1: int, cost2: int) -> int:
        res = 0
        i = 0

        while total - i * cost1 > 0:
            res += (total - i * cost1) // cost2 + 1
            i += 1

        return res + 1 if total - i * cost1 == 0 else res

File: 2240 Number of Ways to Buy Pens and Pencils/2240number-of-ways-to-buy-pens-and-pencils.cpp
class Solution {
public:
    long long waysToBuyPensPencils(int total, int cost1, int cost2) {
        int i = 0;
        long long res = 0;

        while(total - i * cost1 > 0){
            res += long((total - i * cost1) / cost2) + 1;
            i++;
        }
        if (total - i * cost1 == 0){res++;}

        return res;
    }
};